src/ADC.c

```diff
@@ -1,69 +1,380 @@
-#include "stm32f4xx_adc.h"
+/*
+ * ADC.c
+ *
+ *  Created on: Jun 1, 2018
+ *      Author: Lyden
+ */
+
 #include "ADC.h"
+#include "stm32f4xx_rcc.h"
+#include "delay.h"
+#include "UART_Controller.h"
+
+#define GPIO_AFRH_AFRH9_0 (0x00000010)
+#define GPIO_AFRH_AFRH9_1 (0x00000020)
+#define GPIO_AFRH_AFRH9_2 (0x00000040)
+#define GPIO_AFRH_AFRH9_3 (0x00000080)
+
+#define GPIO_AFRH_AFRH8_0 (0x00000001)
+#define GPIO_AFRH_AFRH8_1 (0x00000002)
+#define GPIO_AFRH_AFRH8_2 (0x00000004)
+#define GPIO_AFRH_AFRH8_3 (0x00000008)
+
+#define GPIO_AFRL_AFRL4_0 (0x00010000)
+#define GPIO_AFRL_AFRL4_1 (0x00020000)
+#define GPIO_AFRL_AFRL4_2 (0x00040000)
+#define GPIO_AFRL_AFRL4_3 (0x00080000)
+
+#define GPIO_AFRL_AFRL1_0 (0x00000010)
+#define GPIO_AFRL_AFRL1_1 (0x00000020)
+#define GPIO_AFRL_AFRL1_2 (0x00000040)
+#define GPIO_AFRL_AFRL1_3 (0x00000080)
+
+/* Clock rate of the timer */
+#define TIMER_CLOCK_RATE (67600000)
+
+/* 9 cycles at 67.6MHz gives 7.51MHz which is a bit less than 8MHz */
+#define EXT_CLOCK_PERIOD (9)
+
+/* CONVST PWM with computed period and a on time of 37 cycles */
+#define CONVERSION_TIME (30)
+#define ACQUISITION_TIME (1)
+#define CONVST_PERIOD (((CONVERSION_TIME + ACQUISITION_TIME) * EXT_CLOCK_PERIOD) - 1)
+#define CONVST_DUTY (CONVERSION_TIME * EXT_CLOCK_PERIOD)
+#define CONVST_PSC (0)
+
+/* RD PWM with 50% duty, 3.33 MHz frequency */
```

Comment:
What does RD stand for here?

Reply:
I believe it stands for read. I followed the naming convention from the ADC IC

Reply:
RD is the pin used by the ADC to trigger latching internal data onto the bus. Once it receives a falling edge from RD, that data gets latched, and can be read from the micro. (Assuming CS (Chip Select) is also low)

Reply:
Can you put "RD(Read)" in the comments the first time it you write it in the file?